onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.cc

```diff
@@ -54,6 +54,7 @@ Status SelectorActionTransformer::MatchAndProcess(Graph& graph, Node& node, bool
                                                   const logging::Logger& logger) const {
   Status status = Status::OK();
 
+  const GraphViewer graph_viewer(graph);
```

Comment:
>graph_viewer [](http://example.com/codeflow?start=20&length=12)

perhaps this could be passed in as a parameter?

Reply:
This function will also use the mutable graph in action->Run(..., unless we want to pass both graph and graph_viewer, I think it's ok we only construct graph_viewer once here,

Reply:
MatchAndProcess() is called in a loop and there happens to be a GraphViewer instance declared outside of that loop. if that instance stays valid, perhaps we can just use it? thought it does look funny to pass both Graph& and const GraphViewer&